applications/subghz/views/subghz_frequency_analyzer.c

```diff
@@ -37,71 +64,178 @@ void subghz_frequency_analyzer_set_callback(
+    canvas_draw_str(canvas, 8, 30, buffer);
+    canvas_draw_icon(canvas, 96, 19, &I_MHz_25x11);
+
+    // Last detected frequency
+    canvas_set_font(canvas, FontSecondary);
+    if(model->frequency_last) {
+        snprintf(
+            buffer,
+            sizeof(buffer),
+            "Last: %03ld.%03ld MHz",
+            model->frequency_last / 1000000 % 1000,
+            model->frequency_last / 1000 % 1000);
+    } else {
+        strcpy(buffer, "Last: ---.--- MHz");
+    }
+    canvas_draw_str(canvas, 9, 42, buffer);
+
+    // Buttons hint
+    elements_button_left(canvas, "T-");
+    elements_button_right(canvas, "T+");
 }
 
 bool subghz_frequency_analyzer_input(InputEvent* event, void* context) {
     furi_assert(context);
+    SubGhzFrequencyAnalyzer* instance = context;
+
+    bool need_redraw = false;
 
-    if(event->key == InputKeyBack) {
-        return false;
+    if(event->key == InputKeyBack) return false;
+
+    if(((event->type == InputTypePress) || (event->type == InputTypeRepeat)) &&
+       ((event->key == InputKeyLeft) || (event->key == InputKeyRight))) {
+        // Trigger setup
+        float trigger_level = subghz_frequency_analyzer_worker_get_trigger_level(instance->worker);
+        switch(event->key) {
+        case InputKeyLeft:
+            trigger_level -= TRIGGER_STEP;
+            if(trigger_level < RSSI_MIN) trigger_level = RSSI_MIN;
+            break;
+        default:
+        case InputKeyRight:
+            trigger_level += TRIGGER_STEP;
+            if(trigger_level > RSSI_MAX) trigger_level = RSSI_MAX;
+            break;
+        }
+        subghz_frequency_analyzer_worker_set_trigger_level(instance->worker, trigger_level);
+        FURI_LOG_I(TAG, "trigger = %.1f", (double)trigger_level);
+        need_redraw = true;
+    }
+
+    if(need_redraw) {
+        SubGhzFrequencyAnalyzer* instance = context;
+        with_view_model(
+            instance->view, (SubGhzFrequencyAnalyzerModel * model) {
+                model->rssi_last = instance->rssi_last;
+                model->frequency_last = instance->frequency_last;
+                model->trigger =
+                    subghz_frequency_analyzer_worker_get_trigger_level(instance->worker);
+                return true;
+            });
     }
 
     return true;
 }
 
+uint32_t round_int(uint32_t value, uint8_t n) {
+    // Round value
+    uint8_t on = n;
+    while(n--) {
+        uint8_t i = value % 10;
+        value /= 10;
+        if(i >= 5) value++;
+    }
+    while(on--) value *= 10;
+    return value;
+}
+
 void subghz_frequency_analyzer_pair_callback(void* context, uint32_t frequency, float rssi) {
+    furi_assert(context);
     SubGhzFrequencyAnalyzer* instance = context;
+
     if((rssi == 0.f) && (instance->locked)) {
-        if(instance->callback) {
-            instance->callback(SubGhzCustomEventSceneAnalyzerUnlock, instance->context);
+        notification_message(instance->notifications, &sequence_hw_blink);
```

Comment:
@ClusterM I'm not sure, that's a good idea to blink while scanning. It was convenient when Flipper blinked only when it senses a signal. Now it's harder to detect the moment when I need to pay attention to the screen. I'd rather disable constant blinking in `Read` mode, because it may draw unneeded attention when using in public spaces.

Reply:
https://github.com/ClusterM/flipperzero-firmware/pull/1

Reply:
And my PR with `Read`/`Read RAW` flashing disabled: https://github.com/flipperdevices/flipperzero-firmware/pull/1745